--- ed25519-donna-impl-base.c	2025-01-29 16:56:11.000000000 +0100
+++ src/ed25519-donna/ed25519-donna-impl-base.c	2025-01-30 01:44:33.269350300 +0100
@@ -273,7 +273,7 @@
 #define S2_TABLE_SIZE (1<<(S2_SWINDOWSIZE-2))
 
 /* computes [s1]p1 + [s2]base */
-void ge25519_double_scalarmult_vartime(ge25519 *r, const ge25519 *p1, const bignum256modm s1, const bignum256modm s2) {
+void _ge25519_double_scalarmult_vartime(ge25519 *r, const ge25519 *p1, const bignum256modm s1, const bignum256modm s2) {
 	signed char slide1[256] = {0}, slide2[256] = {0};
 	ge25519_pniels pre1[S1_TABLE_SIZE] = {0};
 #ifdef ED25519_NO_PRECOMP
@@ -469,7 +469,7 @@
 }
 
 /* computes [s1]p1, constant time */
-void ge25519_scalarmult(ge25519 *r, const ge25519 *p1, const bignum256modm s1) {
+void _ge25519_scalarmult(ge25519 *r, const ge25519 *p1, const bignum256modm s1) {
 	signed char slide1[64] = {0};
 	ge25519_pniels pre1[9] = {0};
 	ge25519_pniels pre = {0};
@@ -650,7 +650,7 @@
 	curve25519_set(r->z, 1);
 }
 
-void ge25519_add(ge25519 *r, const ge25519 *p, const ge25519 *q, unsigned char signbit) {
+void _ge25519_add(ge25519 *r, const ge25519 *p, const ge25519 *q, unsigned char signbit) {
 	ge25519_pniels P_ni = {0};
 	ge25519_p1p1 P_11 = {0};
 
@@ -749,7 +749,7 @@
 
 void ge25519_scalarmult_base_wrapper(ge25519 *r, const bignum256modm s){
 #if OPTIMIZE_SIZE_ED25519
-        ge25519_scalarmult(r, &ge25519_basepoint, s);
+        _ge25519_scalarmult(r, &ge25519_basepoint, s);
 #else
 	ge25519_scalarmult_base_niels(r, ge25519_niels_base_multiples, s);
 #endif
